c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2981,11 +2981,24 @@ class DeviceCachingAllocator {
 // go straight to cudaMalloc.  This setting is useful when debugging GPU memory
 // errors, since the caching allocator foils cuda-memcheck.
 bool forceUncachedAllocator() {
-  static bool force_uncached =
+  bool force_uncached =
```

Comment:
why remove static? If remove static, we need to check the env every time. This will increase the overhead, and we don't need to support dynamically update the ENV var, right?

Reply:
So it's for the test purpose?

We can add some helper function, but we shouldn't load the ENV var every time, otherwise, the overhead may be high.

Reply:
Yes, I removed the `static` to support dynamically updating the ENV variable for the test. 

I also share similar concerns about overhead.

Can you say more about the helper function you have in mind?

Reply:
added python API for disabling the caching allocator.  restored the env var to `static`.